src/mpi.jl

```diff
@@ -12,3 +12,17 @@ struct MPICommsContext{D <: AbstractDevice, C} <: AbstractCommsContext
 end
 
 function MPICommsContext end
+
+"""
+    local_communicator(ctx::MPICommsContext)
+
+Internal function to create a new MPI communicator for processes on the same physical node.
+
+Sample Usage:
+```
+ClimaComms.local_communicator(ctx) do local_comm
+    ClimaComms._assign_device(ctx.device, MPI.Comm_rank(local_comm))
+end
+```
+"""
+function local_communicator end
```

Comment:
Should we have a singleton implementation that does nothing?

Reply:
(I'm not sure, but I thought it might be worth discussing)

Reply:
Yeah, I was thinking that it could do nothing, but maybe that's a trap that users could fall into. Maybe better to just not have an implementation.